### DISCUSSION TITLE
DevToys extension : Struggling with nuget package

### DISCUSSION THREAD

**Discussion by Poilaupat** [DevToys-app/DevToys#1280] [Q&A] [answered]
Hi,

I have an idea for a DevToys extension, so I started following the "Getting Started" section. I set up my environment (Windows 11 + Visual Studio 2022 17.10.1 with .NET Desktop workload + environment path configured). However, when I try to create the minimal extension, I get stuck at the step where the resource assembly identifier is created. The `using DevToys.Api` directive triggers an error: `The type or namespace 'DevToys' cannot be found (Are you missing a using directive or an assembly reference?)`.

Here's what I've checked so far:
- The NuGet package is correctly referenced (although I installed it via the VS NuGet Package Manager Interface rather than the command line, but I don't think that matters).
- The target version of my project is .NET 8.
- I tried various solutions proposed on Stack Overflow for similar problems (clean-rebuild, .vs folder delete, uninstall-reinstall of the NuGet package), but nothing worked.

What I've noticed is that the NuGet package doesn't seem to contain a DLL file (I'm not an expert on NuGet packages, so I don't know if this is normal, but I expected to see DevToys.Api.dll). When I compile the project, I can't see DevToys.Api.dll in the output folder. I observe different behavior when I try to add other references (like AutoMapper, for example): the DLLs are deployed in the output folder even if they are not explicitly used in a C# file.

Does this mean that I have to add a DLL reference to DevToys.Api.dll from the installation folder?

Thanks for your help!

**Comment by veler** [comment, marked as answer]
Hi,

Sorry for the inconvenience. Can you please verify that in the CSPROJ, the reference to `DevToys.Api` package is

```xml
    <PackageReference Include="DevToys.Api" />
```

And not

```xml
    <PackageReference Include="DevToys.Api">
      <PrivateAssets>all</PrivateAssets>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
    </PackageReference>
   </ItemGroup>
```

Thanks. Please let me know if it helps.

**Comment by Poilaupat** [comment]
Thanks for your advice.

As you suspected, the DevToys.Api reference in the csproj was similar to the second one you mentioned. I tried reinstalling it both via the VS NuGet Package Manager and the command line, but nothing changed.

Then I updated the csproj manually with `<PackageReference Include="DevToys.Api" Version="2.0.2-preview"/>`, and it worked. I can now see the DLL in the NuGet content, and the compilation is successful.

Thank you very much.
